examples/data-objects/inventory-app/src/inventoryList.ts

```diff
@@ -7,27 +7,42 @@ import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
 import {
 	AllowedUpdateType,
 	ForestType,
-	TypedTreeChannel,
-	TypedTreeFactory,
+	ISharedTree,
+	SharedTreeFactory,
 	typeboxValidator,
 } from "@fluid-experimental/tree2";
 import { IFluidHandle } from "@fluidframework/core-interfaces";
-import { InventoryField, schema } from "./schema";
+import { Inventory, schema } from "./schema";
 
 const treeKey = "tree";
 
-const factory = new TypedTreeFactory({
+const factory = new SharedTreeFactory({
 	jsonValidator: typeboxValidator,
 	forest: ForestType.Reference,
-	subtype: "InventoryList",
 });
 
 export class InventoryList extends DataObject {
-	private _tree: TypedTreeChannel | undefined;
+	private _tree?: ISharedTree;
+	private _inventory?: Inventory;
```

Comment:
This pattern is curious, but it makes sense for right now. We can't mutate the root node. Once we're able to mutate the root node (because `root2` will be a getter/setter, perhaps?) it will make sense to cache the `_view` rather than cache the `_root` (`_inventory`), and then `inventory()` can be have a setter as well.

Reply:
You're right.  We'll want people to cache the 'view' long-term.  I'll make that change.  Thanks.